src/async_impl/client.rs

```diff
@@ -514,6 +535,23 @@ impl Client {
             headers.insert(key, value.clone());
         }
 
+        // Add cookies from the cookie store.
+        if let Some(cookie_store_wrapper) = self.inner.cookie_store.as_ref() {
+            if headers.get(::header::COOKIE).is_none() {
+                let cookie_store = cookie_store_wrapper.read().unwrap();
+                let header = cookie_store
+                    .0
+                    .get_request_cookies(&url)
+                    .map(|c| c.encoded().to_string())
+                    .collect::<Vec<_>>()
+                    .join("; ");
+                if !header.is_empty() {
+                    // TODO: is it safe to unwrap here? Investigate if Cookie content is always valid.
```

Comment:
It looks like since using `encoded`, the value should be percent-encoded, which would make it always valid.

Reply:
yeah just forgot to remove the todo